packages/rocketchat-api/server/v1/groups.js

```diff
@@ -453,42 +454,36 @@ RocketChat.API.v1.addRoute('groups.listAll', { authRequired: true }, {
 RocketChat.API.v1.addRoute('groups.members', { authRequired: true }, {
 	get() {
 		const findResult = findPrivateGroupByIdOrName({ params: this.requestParams(), userId: this.userId });
-		const room = RocketChat.models.Rooms.findOneById(findResult.rid, {fields: {broadcast: 1}});
+		const room = RocketChat.models.Rooms.findOneById(findResult.rid, { fields: { broadcast: 1 } });
 
 		if (room.broadcast && !RocketChat.authz.hasPermission(this.userId, 'view-broadcast-member-list')) {
 			return RocketChat.API.v1.unauthorized();
 		}
 
 		const { offset, count } = this.getPaginationItems();
-		const { sort } = this.parseJsonQuery();
-
-		let subscriptionSort = {};
-
-		if (!Match.test(sort, Object) || !Match.test(sort.username, Number)) {
-			subscriptionSort = {'u.username': 1};
-		} else {
-			subscriptionSort = {'u.username': sort.username};
-		}
+		const { sort = {} } = this.parseJsonQuery();
 
-		const members = RocketChat.models.Subscriptions.findByRoomId(findResult.rid, {
-			fields: {u: 1},
-			sort: subscriptionSort,
+		const subscriptions = RocketChat.models.Subscriptions.findByRoomId(findResult.rid, {
+			fields: { 'u._id': 1 },
+			sort: { 'u.username': sort.username != null ? sort.username : 1 },
 			skip: offset,
 			limit: count
-		}).fetch()
-			.filter(s => s.u && s.u._id && s.u.username)
-			.map(s => s.u.username);
+		});
+
+		const total = subscriptions.count();
```

Comment:
Total will count the users without username???

Reply:
sure, I think this code has the same behavior
```javascript
{
	total: RocketChat.models.Subscriptions.findByRoomId(findResult.rid).count()	
}
```

Reply:
I don't think so, you are including the count and skip right?

Reply:
`count()` ignores `limit` and `skip`